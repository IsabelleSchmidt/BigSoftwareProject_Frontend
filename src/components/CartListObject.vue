<template>
<div id="line">
    <div class ="productobject">
        <div class="picture">
                <img v-bind:src="product.allPictures[0].path" alt="Picture" id="pic">
        </div>
        <div class="information">
            <ul>
                <li id="prName">{{product.name}}</li>
                <li id="prPrice">{{product.price}} €</li> 
                <li id="prNr">
                    <span>Pnr: {{product.articlenr}} </span>
                </li>
                <li id="inTotal"> 
                    <span>Gesamtpreis: {{p}} €</span>
                </li> 
                <input :value="amount" @change="amChange($event.target.value)" min="1" max="20" type="number" id="amount">
            </ul>
        </div>
        <div class="close">
            <button id="delete" @click="trash()">
                <img src="../assets/closeImg.png">
            </button>
        </div>
    </div>
</div>
    
</template>
<script lang="ts">
import '@/service/Product'
import { defineComponent, onMounted, ref, computed} from 'vue'

export default defineComponent({
    name: "CartListObject",
    props: {
        product:{
            type: Object,
            default: ()=> ({}),
        }
        
    }, 
     
    setup(props, context) {
        const amount = ref(1);
        const Pprice = ref(props.product.price);
        const umrechnungsfaktor = Math.pow(10,2);


        const p = computed(() => {
            return Pprice.value;
        }); 
        
         onMounted(async () => {

             const aList = localStorage.getItem('amount');
             if(aList){
                const list = JSON.parse(aList);  

                for(let i = 0; i< aList.length; i++){
                    if(list[i].name == props.product.name ){
                        amount.value =  list[i].menge ; 
                        break;
                    } 
                } 
                Pprice.value = Math.round((amount.value * props.product.price)*umrechnungsfaktor)/umrechnungsfaktor; 
                console.log("Gesamt: " + Pprice.value);
             }
                   
        });
        function amChange(am: number): void{

            const aList = localStorage.getItem('amount');
             if(aList){
                const list = JSON.parse(aList); 

                for(let i = 0; i< aList.length; i++){
                    if(list[i].name == props.product.name ){
                        amount.value = am; 
                        list[i].menge = amount.value;
                        break; 
                    } 
                }

                Pprice.value=Math.round((amount.value * props.product.price)*umrechnungsfaktor)/umrechnungsfaktor; 
                localStorage.setItem('amount', JSON.stringify(list));

            }    
            context.emit("amount-change", am);
        } 

       function trash(): void {
           context.emit("delete-product", props.product);
       } 
       return {
           amount,
           p,
           amChange,
           trash
       };
    },
});
</script>
<style lang = "scss">
ul{
    list-style: none;
} 
.information li{
    padding-top: 2.5%;
    padding-bottom: 2.5%;
    text-align: left;
} 
#prName{
    font-size: 16px;
    font-weight: bold;
    padding-bottom: 0%;
} 
#prPrice{
    font-size: 16px;
    padding-top: 0%;
} 
#prNr{
    font-size: 12px;
    padding-bottom: 0%;
} 
#inTotal{
    font-size: 12px;
    padding-top: 0%;
} 
#amount{
    float: left;
    font-size: 14px;
} 
    .productobject{
        margin-left: 2em;
        margin-right: 2em;
        margin-top: 1.5em;
        margin-bottom: 1.5em;
        min-width: 35em;
        float: left;
    } 
    #delete{
        border-style: none;
        background: none;
        &:focus{
            outline: none;
        } 
    } 
   
    #pic{
        float: left;
        width: 5em;
    } 
     #line{
        border-bottom: 2px solid #d4d4d4;
    }  

    .information{
        float: left;
        margin-left: 5%;
    } 

    .close{
        float: right;
    } 
</style>