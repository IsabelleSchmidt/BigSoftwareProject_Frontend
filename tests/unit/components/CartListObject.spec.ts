import { shallowMount, VueWrapper} from '@vue/test-utils'
	
import CartListObject from '@/components/CartListObject.vue'
	
describe('CartListObject', () => {
	
    let cmp: VueWrapper<any>
	
	
    beforeEach(() => {
	
        cmp = shallowMount(CartListObject)
	
    })
    test('has the expected structur', () => {
        expect(cmp).toMatchSnapshot()
    })

    // test('If html contains the productobject, picture and all important informations', () => {
    //     expect(cmp.html()).toContain('productobject')
    //     expect(cmp.html()).toContain('picture')
    //     expect(cmp.html()).toContain('information')
    //     expect(cmp.html()).toContain('prName')
    //     expect(cmp.html()).toContain('prPrice')
    //     expect(cmp.html()).toContain('prNr')
    //     expect(cmp.html()).toContain('inTotal')
    //     expect(cmp.html()).toContain('amount')
    // })
})