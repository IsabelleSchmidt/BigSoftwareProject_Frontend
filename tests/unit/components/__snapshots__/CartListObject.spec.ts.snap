// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartListObject has the expected html structur 1`] = `undefined`;
